src/zip.cpp

```diff
@@ -131,26 +128,26 @@ namespace zip
     pfileIn = fopen(zi.filename.c_str(), "rb"); // open ZIP file for reading
     if (pfileIn == nullptr) {
       LOG_ERROR("Couldn't open zip file for reading: " << zi.filename);
+      fclose(*pfileOut);
```

Comment:
Don't you rather want to do the same (use a scope_exit) for pFileOut?

Reply:
pFileOut is different, it s "inout" parameter still need to be used up to `file_load`. If I put it in the scope_exit too, it would become invalid once out of scope of extract.